### PR TITLE
Merge to upstream 798bfd7df

### PR DESCRIPTION
**Pull request by cferry-AMD**
This is the latest upstream commit that can merge in here without any conflicts or failures. Passing CI requires pinning stable `torch` to 2.3.1 that was the stable version at the time of that commit.

### CODE REVIEW

File: build_tools/ci/install_python_deps.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ case $torch_version in
     ;;
   stable)
     echo "::group::installing stable torch"
-    python3 -m pip install --no-cache-dir torch torchvision --index-url https://download.pytorch.org/whl/cpu
+    python3 -m pip install --no-cache-dir torch==2.3.1 torchvision --index-url https://download.pytorch.org/whl/cpu
```
Comment (mgehre-amd) [inline build_tools/ci/install_python_deps.sh]:
```suggestion
    python3 -m pip install --no-cache-dir -r stable-requirements.txt
```

**Review by mgehre-amd — APPROVED** [review]

File: build_tools/ci/install_python_deps.sh (hunk 1 above)
Comment (cferry-AMD) [reply build_tools/ci/install_python_deps.sh]: Thanks!

File: build_tools/ci/install_python_deps.sh [hunk 2]
```diff
@@ -19,7 +19,7 @@ case $torch_version in
     ;;
   stable)
     echo "::group::installing stable torch"
-    python3 -m pip install --no-cache-dir torch torchvision --index-url https://download.pytorch.org/whl/cpu
+    python3 -m pip install --no-cache-dir torch torchvision --index-url -r $repo_root/stable-requirements.txt
```
Comment (mgehre-amd) [inline build_tools/ci/install_python_deps.sh]:
```suggestion
    python3 -m pip install --no-cache-dir -r $repo_root/stable-requirements.txt
```

**Comment by cferry-AMD** [comment]
I squashed my additional commits into a single one for the history to be clean upon merge into our branch.

**Pull request merged**